server/src/inmantals/server.py

```diff
@@ -64,9 +65,149 @@ def __init__(self, message: str) -> None:
+        """
+        return self.folder_uri.path
+
+    def create_tmp_project(self) -> str:
+        """
+        When working on a module, an inmanta project is created in a temporary directory.
+        """
+        self.inmanta_project_dir = tempfile.TemporaryDirectory()
+        logger.debug(f"Temporary project created at {self.inmanta_project_dir.name}.")
+
+        os.mkdir(os.path.join(self.inmanta_project_dir.name, "libs"))
+
+        install_mode = module.InstallMode.master
+
+        v2_metadata_file: str = os.path.join(self.get_folder_path(), module.ModuleV2.MODULE_FILE)
+        v1_metadata_file: str = os.path.join(self.get_folder_path(), module.ModuleV1.MODULE_FILE)
+
+        module_name: Optional[str] = None
+        libs_folder = os.path.join(self.inmanta_project_dir.name, "libs")
+
+        if os.path.exists(v1_metadata_file):
+            mv1 = module.ModuleV1(project=None, path=self.get_folder_path())
+            module_name = mv1.name
+            os.symlink(self.get_folder_path(), os.path.join(libs_folder, module_name), target_is_directory=True)
+
+        elif os.path.exists(v2_metadata_file):
+            mv2 = module.ModuleV2(project=None, path=self.get_folder_path(), is_editable_install=True)
+            module_name = mv2.name
+            # Install this v2 module in editable mode in the active venv.
+            env.process_env.install_from_source([env.LocalPackagePath(path=self.get_folder_path(), editable=True)])
+
+        if not module_name:
+            error_message: str = (
+                "The Inmanta extension only works on projects and modules. "
+                f"Please make sure the folder opened at {self.get_folder_path()} is a valid "
+                "[project](https://docs.inmanta.com/inmanta-service-orchestrator/latest/model_developers/project_creation.html)"
+                " or "
+                "[module](https://docs.inmanta.com/inmanta-service-orchestrator/latest/model_developers/module_creation.html)"
+            )
+            self.cleanup()
+            raise InvalidExtensionSetup(error_message)
+
+        repos = self.handler.get_setting("inmanta.repos", self, "")
+
+        logger.debug(f"project.yaml created at {os.path.join(self.inmanta_project_dir.name, 'project.yml')} repos={repos}.")
+        with open(os.path.join(self.inmanta_project_dir.name, "project.yml"), "w+") as fd:
+            metadata: typing.Mapping[str, object] = {
+                "name": "Temporary project",
+                "description": "Temporary project",
+                "repo": yaml.safe_load(repos),
+                "modulepath": "libs",
+                "downloadpath": "libs",
+                "install_mode": install_mode.value,
+            }
+            yaml.dump(metadata, fd)
+
+        with open(os.path.join(self.inmanta_project_dir.name, "main.cf"), "w+") as fd:
+            fd.write(f"import {module_name}\n")
+
+        return self.inmanta_project_dir.name
+
+    def __repr__(self):
```

Comment:
This is not what this method is for. It is a fine and likely not widely-known line so it's an easy mistake to make. Generally, `repr` is meant to be a structured, unambiguous, and ideally machine-consumable representation. `str` on the other hand is for human consumption and has no strict requirements. See the docs for [`__repr__`](https://docs.python.org/3/reference/datamodel.html#object.__repr__) and `__str__` below it.